.gitlab-ci.yml

```diff
@@ -6,6 +6,11 @@ variables:
   DOCKER_HUB_USERNAME: 'menderbuildsystem'
   DOCKER_BUILDKIT: 1
 
+.tag_n_push_to_gitlab_registry: &tag_n_push_to_gitlab_registry
+  - docker pull ${CI_REGISTRY_IMAGE}:${CONTAINER_TAG}-${CI_PIPELINE_ID}
+  - docker tag ${CI_REGISTRY_IMAGE}:${CONTAINER_TAG}-${CI_PIPELINE_ID} ${CI_REGISTRY_IMAGE}:${CONTAINER_TAG}
+  - docker push ${CI_REGISTRY_IMAGE}:${CONTAINER_TAG}
```

Comment:
just as a general question in light of the potential usage for the multiarch builds (and unrelated to this PR): do you think this would tag & push the image for all architectures? I think the `pull` will just pull for the local architecture and thus the tag & push might be limited to the local architecture..
IMO it would be nice if the registry would notice that the tag change would apply to all related architectures, but I don't think registries are smart enough...

Reply:
TBH I don't know, not an expert on multiarch builds. AFAIK we need build hosts with qemu for supporting multiarch, then we'll test it.
I can guess that we can create a for cycle to pull containers for every arch, at worst...